middle_end/flambda2/types/flambda2_types.mli

```diff
@@ -646,6 +648,9 @@ val prove_unique_fully_constructed_immutable_heap_block :
 
 val prove_is_int : Typing_env.t -> t -> bool proof_of_property
 
+(* Either a tagged integer or a null poitner. *)
+val prove_is_immediate : Typing_env.t -> t -> bool proof_of_property
```

Comment:
I don't think the comment is enough: changing this to `prove_is_not_a_pointer` would be a safe alternative until we sort out naming issues.